fpdf/html.py

```diff
@@ -311,6 +312,29 @@ def __init__(
         self.td_th = None  # becomes a dict of attributes when processing <td>/<th> tags
         # "inserted" is a special attribute indicating that a cell has be inserted in self.table_row
 
+        _old_default_colors: dict[str, tuple[int, int, int]] = {
+            "link": (0, 0, 255),
+            "li": (190, 0, 0),
+            "blockquote": (100, 0, 45),
+            "headings": (150, 0, 0),
+        }
+        if not element_colors:
+            element_colors = {}
+
+        self.link_color: tuple[int, int, int] = element_colors.get(
+            "link", _old_default_colors["link"]
+        )
+        self.li_color: tuple[int, int, int] = element_colors.get(
+            "li", _old_default_colors["li"]
+        )
+        self.blockquote_color: tuple[int, int, int] = element_colors.get(
+            "blockquote", _old_default_colors["blockquote"]
+        )
+
+        self.headings_color: tuple[int, int, int] = element_colors.get(
+            "headings", _old_default_colors["headings"]
+        )
+
```

Comment:
```suggestion
        self.element_colors = {**DEFAULT_ELEMENT_COLORS, **element_colors}
```
I think you could simply store a single `self.element_colors` property.

And the default values would be better defined as a module constant.

Reply:
May I ask the reason you would want to increase the scope of the defaults to a module level?

Whilst my suggested code may cover more lines (in part due to Black's archaic default line length), it only takes from the dictionary what is needed and leaves everything else alone. I say this as I have been using code similar to this with your library for sometime and the passed dict is a colour scheme with more entries than what you see here and which are not related to HTML. For example background colour, text colour, specific hi contrast colours etc. to use in other parts when generating a pdf.

I originally used instance variables instead a member dict as it suited the surrounding code. Whilst my suggested code maybe a few lines longer, it is explict and both easy to read and understand from looking at a single location.

Background.
I am generating student code reports for assignments and have a number of different colour schemes due to medical and preference for students.

Reply:
If the concern was the number of lines added to the __init__ method, how about the revised refactor to a member method?

Reply:
> May I ask the reason you would want to increase the scope of the defaults to a module level?

For code readability.
This is currently a code style standard in `fpdf2` to define constants at the top level of modules.
This also a very common choix from many Python code style, including Google one:
https://google.github.io/styleguide/pyguide.html#254-decision
> Module-level constants are permitted and encouraged. [...] Constants must be named using all caps with underscores.

> Whilst my suggested code maybe a few lines longer, it is explict and both easy to read and understand from looking at a single location.

This is debatable 🙂
IMHO, "hiding" constants inside functions does not produce very clean code.
But in the end, I'm willing to admit that's its mostly a matter of taste & habits.
However, what is important I think, and more "objective", is that **a contribution to a code base must try to conform to the existing code style**.
If I'd contribute to your project, and you'd explain to me that your code style recommends to define default values inside functions, possibly at their beginning, I would adapat my code to conform to your existing code style.
Do you agree with that @dmail00?

Reply:
> If the concern was the number of lines added to the init method, how about the revised refactor to a member method?

This is not the concern 😅
The concern is more about the number of lines added to `fpdf2` **by this PR**.
In that regard, introducing a new method will not make a difference.

But overall, the concern is about code clarity and maintainability.
I'm actually more concerned about introducing 4 new attributes to this class than to add 20 lines of extra code 😅

Reply:
But it is only global state because you increased the scope to global.

The request is LGPL you are free to do with it how you please.

Reply:
> The request is LGPL you are free to do with it how you please.

Are you implying that you do not wish to work on this PR anymore?